src/orchestration/KubernetesClient.js

```diff
@@ -40,6 +40,26 @@ class KubernetesClient {
    * @returns {Promise<EngineContainerSpec[]>} A promise to a list of engine container specs.
    */
   static async listEngines() {
+    const replicaMap = new Map();
+    try {
```

Comment:
Any reason why we're doing these sequentially? AFAIK they don't use any data from previous calls and can be parallelized?

E.g.

```
const replicas = kubeHttpGet('/apis/apps/v1/replicasets');
const deployments = kubeHttpGet('/apis/apps/v1/deployments');
const result = await Promise.all([replicas, deployments]);
// result[0] = replicas, etc...
```

Reply:
No you are right, we should definitely parallelize them :)